### ISSUE TITLE
Have a "welcoming newcomers" page

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/team-compass#29]
I ran across this link in a GSOC email thread:

https://publiclab.github.io/community-toolbox/

I think it is *super cool* to put this much work into creating an inviting space for contributors. How do folks think about implementing something like this?

**Comment by willingc** [comment]
Would be fantastic! @jzf2101 this looks like something you would interest you :smile:

**Comment by betatim** [comment]
I like it. Especially the idea of "first-timers-only" as a tag that is different from "help-wanted". Can we introduce that as of now already?

**Comment by jzf2101** [comment]
`good first issue` is the traditional tag

**Comment by choldgraf** [comment]
> Can we introduce that as of now already?

I'm all for it!

**Comment by jzf2101** [comment]
We should probably still use `good first issue` since it builds on the features built in GitHub for new contributors

**Comment by choldgraf** [comment]
looks like we already have this label (on some repos at least), but it hasn't been used much :-/

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

**Comment by choldgraf** [comment]
but I'm +1 with @jzf2101 we shouldn't change convention if there already is one!

**Comment by jzf2101** [comment]
@choldgraf see https://github.com/jupyter/help/blob/master/CONTRIBUTING.md as a reference @willingc suggests we add repo2docker as well

**Comment by betatim** [comment]
The thing I learnt from the guide choldgraf linked was to have a label for issues that are exclusively for first timers. Calling it "first-timers-only" conveys to everyone that you can only work on this if this is your first contribution, after that you need to switch to "help wanted" (or some such).

What I've seen happen is that someone who isn't a first timer comes and "let me quickly fix those easy to fix issues". On the one hand that is great, on the other the reason those issues are there is to encourage more people to join. They have to be simple, fun, and easy to do for an experienced contributor. Which means experienced people love closing them...instead to help newcomers we should commit to "these are off-limits to everyone else, if you work on this someone will help/teach/mentor you and answer all your 'silly' questions, git screw ups will be forgiven, etc". Basically closing this issue by proxy is going to take much longer than doing it yourself but in exchange we hopefully gain a new contributor.

TL;DR: I think there is a difference between general "help wanted" and specific "first timers" issues. I had never thought about it this way before reading the GSOC link but think it would be a good thing to try out in terms of on-boarding people.

An alternative is to establish the same rules of the "first timers only" issues for "help wanted". I think this would be harder and more negative as they already have a well established meaning and we'd have to go around closing PRs with "sorry but you should have read CONTRIBUTING.md which tells you that you shouldn't work on these" which is :-/

**Comment by jzf2101** [comment]
@betatim I will make a similar point - which is I think that a lot of people who have made their first PR still has a barrier to their second and third because they lack confidence in the codebase.

**Comment by choldgraf** [comment]
I am happy with either approach for labels - to me the more important thing is that such a page exists in the first place :-)  anybody should feel free to make a PR and we can move from there! Else I'll try to get to this later when I have a moment!

@jzf2101 that is an intriguing question! Shouldn't be too hard to scrape that dataset from github w/ `pygithub` or somesuch thing.

**Comment by jzf2101** [comment]
Yeah I was planning on doing that let's see if @staeiou and @nellev are interested

**Comment by jzf2101** [comment]
This is related to https://github.com/jupyter/jupyter.github.io/issues/307
